lib/contextual_logger.rb

```diff
@@ -106,7 +106,7 @@ def add(arg_severity, arg1 = nil, arg2 = nil, **context)   # Ruby will prefer to
         if arg1.nil?
           if block_given?
             message = yield
-            progname = arg2 || context[:progname] || @progname
+            progname = arg2 || context.delete(:progname) || @progname
```

Comment:
@jebentier I added this back as I had it, so that the tunneled `progname` doesn't live on in the context hash.
Usually I'd avoid `delete` since that mutates the `context` here. But it's really succinct, and performs best...and I figured no one else but our code above would be putting a `:progname` key into the context and calling `add` in the block form with a context hash! LMK if you disagree.

Reply:
I think this is okay for now.  Maybe something to not in the README is that through this implementation you're able to specify `progname` using two different method (flip it to a feature 😉 )

`logger.info('request') { message }` and `logger.info(progname: 'request') { message }` will pass progname down to the formatter.

Reply:
@jebentier Can you take a look at the README restructure here: https://github.com/Invoca/contextual_logger/pull/34/commits/17b8f29208d01632b88ad5e41b01fce96236038f